extensions/markdown-language-features/src/languageFeatures/documentLinkProvider.ts

```diff
@@ -207,9 +213,16 @@ async function findCode(document: SkinnyTextDocument, engine: MarkdownEngine): P
 	return { multiline, inline };
 }
 
-function isLinkInsideCode(code: CodeInDocument, link: MdLink) {
-	return code.multiline.some(interval => link.source.hrefRange.start.line >= interval[0] && link.source.hrefRange.start.line < interval[1]) ||
-		code.inline.some(position => position.intersection(link.source.hrefRange));
+/**
+ * Check whether the given {@link linkHrefRange} is inside any code sections.
+ * Ignore those inside code sections.
+ * @param code
```

Comment:
[nit] Only add doc comments if there's something worth documenting. In this case you should remove all the `@param` and `@returns` because they don't provide any value